### DISCUSSION TITLE
Linking to a config.js

### DISCUSSION THREAD

**Discussion by seanturner026** [renovatebot/renovate#31742] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi,

This should be quite easy. 

I'm running renovate in a monorepo that consists of ArgoCD ApplicationSets. I'm using kustomize overlays to declare separate environments, so I want renovate to open separate PRs for dependency upgrades so that staging can be upgrade before production.

I'm trying to use a `config.js` as I'll have multiple environments / clusters that I'm working with. I've got the below:

renovate.json https://github.com/seanturner026/argocd-applicationset/blob/65dc84812d44fbbbb007ea1ed4a1ea2e196641f3/renovate.json
renovate.js https://github.com/seanturner026/argocd-applicationset/blob/8676dbea34ac2fb8062cd67f9b3089ee31597343/renovate.js

A Renovate-managed Github Issue is complaining that I can't use `local>renovate.js` the way I am currently. https://github.com/seanturner026/argocd-applicationset/issues/31

1. How should I get around this? 
1. Is there a solution I can use that would also work for a private repository?
1. Is it possible to combine renovate.json and renovate.js into a single file? config.js?

Thank you very much!!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
1. The way to linked the preset is wrong, it should be:

```js
"extends": ["local>seanturner026/argocd-applicationset:renovate.js"]
```

2. Only `json` files are allowed as presets. So you cannot link `renovate.js`

3. Sorry, I don't understand your needs very well..so cannot comment on the appraoch you need to take. Let's wait for a maintainers comment on that.

**Reply by seanturner026** [reply]
Cheers, thanks for the help. This unfortunately didn't solve the issue. Not a hassle however, mainly commenting so that it's more clear what the next steps are.

https://github.com/seanturner026/argocd-applicationset/issues/31

> 3. Is it possible to combine renovate.json and renovate.js into a single file? config.js?
> 3. Sorry, I don't understand your needs very well..so cannot comment on the appraoch you need to take. Let's wait for a maintainers comment on that.

I was asking if it's possible to just use config.js. But looking a bit more at the docs it looks like I definitely need a renovate.json
